api/src/wfl/module/xx.clj

```diff
@@ -95,30 +95,35 @@
 
 ;; visible for testing
 (defn submit-workload! [{:keys [uuid workflows] :as workload}]
-  (letfn [(update-workflow [workflow cromwell-uuid]
-            (assoc workflow :uuid cromwell-uuid
-                            :status "Submitted"
-                            :updated (OffsetDateTime/now)))]
-    (let [path        (wdl/hack-unpack-resources-hack (:top workflow-wdl))
-          environment (get-cromwell-environment workload)]
-      (logr/infof "submitting workload %s" uuid)
-      (mapv update-workflow
-        workflows
-        (cromwell/submit-workflows
-          environment
-          (io/file (:dir path) (path ".wdl"))
-          (io/file (:dir path) (path ".zip"))
-          (map (comp (partial cromwellify-inputs environment) :inputs) workflows)
-          (util/make-options environment)
-          (merge cromwell-labels {:workload uuid}))))))
+  (let [path                 (wdl/hack-unpack-resources-hack (:top workflow-wdl))
+        environment          (get-cromwell-environment workload)
+        ;; Batch calls have uniform options, so we must group by options to submit
+        workflows-by-options (seq (group-by :workflow_options workflows))]
+    (letfn [(update-workflow [workflow cromwell-uuid]
+              (assoc workflow :uuid cromwell-uuid
+                              :status "Submitted"
+                              :updated (OffsetDateTime/now)))
+            (submit-workflows-by-options [[options associated-workflows]]
```

Comment:
what's an `associated-workflow`?

Reply:
A collection of workflows that all have the same options. This is an xx-specific things to allow us to easily implement the batch endpoint in the future. (Ed and I talked about this offline for a bit but for other folks tuning in, the xx module is "batch endpoint ready" and this behavior of grouping workflows by their options before making batch calls keeps it that way)

Reply:
At risk of being verbose I changed the signature here to be `(submit-workflows-by-options [[options workflows-with-those-options]]` which I think might be a bit more clear

Reply:
I really don't think you need the verbose names in this function - it's pretty clear from the usage. If anything, introducing new names can just cause confusion.

Reply:
Alright, I've given it another shot and renamed it to `ws` since `workloads` is already defined by the outer scope, I didn't want to shadow.